herddb-core/src/main/java/herddb/core/TableManager.java

```diff
@@ -1408,6 +1406,16 @@ private void applyTruncate() throws DataStorageManagerException {
 
     }
 
+    private void unloadAllPagesForTruncate() {
```

Comment:
I was thinking about a possible race condition issue in this code with concurrent flush for newpages:

Tread A start a table delete on T, loops and start to remove pages.
Concurrently Thread B add a page to policy and a "newpage" X of T is selected for unload before that Tread A can truly remove X from policy.
Thread A finish page removal from policy and delete page X data from disk and table directory
Thread B attempt X flush but it fails because there isn't a directory anymore.

It is improbable but still possible.
We should add a "virtual" flush when removing newpages (just to set them ad not writable anymore): another flush will just be avoided by standard code.